io-engine/tests/snapshot_nexus.rs

```diff
@@ -0,0 +1,159 @@
+pub mod common;
+
+use once_cell::sync::OnceCell;
+
+use common::compose::MayastorTest;
+
+use common::compose::{
+    rpc::v1::{
+        bdev::ListBdevOptions,
+        pool::CreatePoolRequest,
+        replica::CreateReplicaRequest,
+        GrpcConnect,
+    },
+    Builder,
+    ComposeTest,
+};
+
+use io_engine::{
+    bdev::{device_create, device_open},
+    core::{MayastorCliArgs, SnapshotParams},
+    subsys::{Config, NvmeBdevOpts},
+};
+
+use std::str;
+use uuid::Uuid;
+
+static MAYASTOR: OnceCell<MayastorTest> = OnceCell::new();
+
+/// Get the global Mayastor test suite instance.
+fn get_ms() -> &'static MayastorTest<'static> {
+    MAYASTOR.get_or_init(|| MayastorTest::new(MayastorCliArgs::default()))
+}
+
+fn replica_name() -> String {
+    "volume1".to_string()
+}
+
+fn replica_uuid() -> String {
+    "65acdaac-14c4-41d8-a55e-d03bfd7185a4".to_string()
+}
+
+fn pool_uuid() -> String {
+    "6e3c062c-293b-46e6-8ab3-ff13c1643437".to_string()
+}
+
+/// Launch a containerized I/O agent with a shared volume on it.
+async fn launch_instance() -> (ComposeTest, String) {
+    common::composer_init();
+
+    Config::get_or_init(|| Config {
+        nvme_bdev_opts: NvmeBdevOpts {
+            timeout_us: 2_000_000,
+            keep_alive_timeout_ms: 5_000,
+            transport_retry_count: 2,
+            ..Default::default()
+        },
+        ..Default::default()
+    })
+    .apply();
+
+    let test = Builder::new()
+        .name("cargo-test")
+        .network("10.1.0.0/16")
+        .unwrap()
+        .add_container_dbg("ms1")
+        .with_clean(true)
+        .build()
+        .await
+        .unwrap();
+
+    let conn = GrpcConnect::new(&test);
+
+    let mut ms1 = conn.grpc_handle("ms1").await.unwrap();
+
+    ms1.pool
```

Comment:
We have wrappers for some v1 APIs to facilitate writing test code, and to make code more compact.